src/ol/source/Google.js

```diff
@@ -238,10 +252,48 @@ class Google extends TileImage {
     const timeout = Math.max(expiry - Date.now() - 60 * 1000, 1);
     this.sessionRefreshId_ = setTimeout(() => this.createSession_(), timeout);
 
+    this.setAttributions(this.fetchAttributions.bind(this));
     // even if the state is already ready, we want the change event
     this.setState('ready');
   }
 
+  async fetchAttributions(frameState) {
+    if (
+      frameState.viewHints[ViewHint.ANIMATING] ||
+      frameState.viewHints[ViewHint.INTERACTING]
+    ) {
+      return this.previousViewportAttribution_;
+    }
+    const transform = getTransformFromProjections(
+      this.getProjection(),
+      getProjection('EPSG:4326'),
+    );
+    const epsg4326Extent = applyTransform(frameState.extent, transform);
```

Comment:
Using
```js
import { transform as transformExtent } from `ol/extent.js`;
```
you can simplify this to
```suggestion
    const epsg4326Extent = transformExtent(frameState.extent, this.getProjection(),
      getProjection('EPSG:4326'));
```

Reply:
The Bing issues in #14526 also apply here but are easier to fix

To handle wrapping the longitudes should be normalized https://developers.google.com/maps/documentation/tile/2d-tiles-overview#viewport_info_request (or set to-180 +180 if more than one world width is visible).

To handle reprojection the extent should be transformed to EPSG:4326 from the view projection instead of the source projection, in which case the resolution used to calculated the zoom would need to be obtained using `ol/reproj#calculateSourceExtentResolution`.

Reply:
@mike-000 I think reprojection is beyond the scope of this pull request, and I doubt it is allowed to reproject a Google map image.

Reply:
Good point about the -180...+180 range. The easiest way to achieve this is to use `toLonLat()` instead of `transformExtent`:
```js
const [west, south] = toLonLat(getBottomLeft(frameState.extent), frameState.viewState.projection);
const [east, north] = toLonLat(getTopRight(frameState.extent), frameState.viewState.projection);
```

Reply:
@ahocevar your suggestion don't seem to work. There is no transform function in ol/extent.js, only applyTransform (i copied the code from Bing Maps source)

Reply:
Where is the toLonLat function from ? I can't find it in the source code

Reply:
transformExtent is in proj, nevermind

Reply:
toLonLat as well.

Reply:
ok, got a bit confused as "transform as transformExtent" doesn't work when using proj, because the first argument is a coordinates, so it was projecting only the first 2 coordinates of the extent. Ended up using toLatLon to get the two corners like you suggested, should be good now